indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -331,7 +331,9 @@ public static boolean runJobs(List<Jobby> jobs, HadoopDruidIndexerConfig config)
         Path workingPath = config.makeIntermediatePath();
         log.info("Deleting path[%s]", workingPath);
         try {
-          workingPath.getFileSystem(injectSystemProperties(new Configuration())).delete(workingPath, true);
+          if( workingPath.getFileSystem(injectSystemProperties(new Configuration())).delete(workingPath, true) ){
+            existing.clear();
```

Comment:
i believe, existing.clear() does not matter at this point because jvm exits after this, workingPath cleanup is required because it is persisted data.